src/core/linalg/src/sparse/4C_linalg_utils_sparse_algebra_print.cpp

```diff
@@ -228,4 +230,18 @@ void Core::LinAlg::print_map_in_matlab_format(
   }
 }
 
+
+/*----------------------------------------------------------------------*/
+/*----------------------------------------------------------------------*/
+Core::LinAlg::SparseMatrix Core::LinAlg::read_matrix_market_file_as_sparse_matrix(
+    const std::string& filename, MPI_Comm comm)
+{
+  Epetra_CrsMatrix* crs_marix;
+
+  ASSERT_EPETRA_CALL(EpetraExt::MatrixMarketFileToCrsMatrix(
+      filename.c_str(), Core::Communication::as_epetra_comm(comm), crs_marix));
+
+  return SparseMatrix(Core::Utils::shared_ptr_from_ref(*crs_marix), DataAccess::Copy);
```

Comment:
We need to make sure that the crs_matrix is deleted once it gets out of scope. Here, we can directly create a shared-pointer and share the data with the returned object.


```suggestion
  return SparseMatrix(std::shared_ptr<Epetra_CrsMatrix>(crs_marix), DataAccess::Share);
```